hudi-common/src/main/java/org/apache/hudi/metrics/prometheus/PrometheusReporter.java

```diff
@@ -101,12 +148,86 @@ public void report() {
 
   @Override
   public void stop() {
-    collectorRegistry.unregister(metricExports);
-    HTTPServer httpServer = PORT_TO_SERVER.remove(serverPort);
-    if (httpServer != null) {
-      httpServer.stop();
+    if (stopped.get()) {
+      LOG.debug("PrometheusReporter.stop() called for port {} but already stopped", serverPort);
+      return;
+    }
+    
+    try {
+      synchronized (PrometheusReporter.class) {
+        stopped.set(true);
+        LOG.debug("PrometheusReporter.stop() called for port {}", serverPort);
+        unregisterMetricExports();
+        removeFromExportsTracking();
```

Comment:
I see we are calling below 
```
PrometheusServerState serverState = PORT_TO_SERVER_STATE.get(serverPort);
```
in some of these methods. 
why not call this at L 159, and pass the `PrometheusServerState` as an arg to each of these methods. 


not a blocking comment. but good to fix it